pkg/analytics/instrumentation_collector.go

```diff
@@ -0,0 +1,243 @@
+package analytics
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/snyk/error-catalog-golang-public/snyk_errors"
+
+	api "github.com/snyk/go-application-framework/internal/api/clients"
+
+	"github.com/snyk/go-application-framework/pkg/local_workflows/json_schemas"
+	"github.com/snyk/go-application-framework/pkg/networking"
+)
+
+const (
+	Success Status = "success"
+	Failure Status = "failure"
+)
+
+type Status string
+
+type InstrumentationCollector interface {
+	SetUserAgent(ua networking.UserAgentInfo)
+	SetInteractionId(id string)
+	SetTimestamp(t time.Time)
+	SetDuration(duration time.Duration)
+	SetStage(s string)
+	SetType(t string)
+	SetCategory(c []string)
+	SetStatus(s Status)
+	SetTestSummary(s json_schemas.TestSummary)
+	SetTargetId(t string) // maybe use package-url library and types
+	AddError(err error)
+	AddExtension(key string, value interface{})
+}
+
+var _ InstrumentationCollector = (*instrumentationCollectorImpl)(nil)
+
+func NewInstrumentationCollector() InstrumentationCollector {
+	return &instrumentationCollectorImpl{}
+}
+
+type instrumentationCollectorImpl struct {
+	userAgent           networking.UserAgentInfo
+	interactionId       string
+	timestamp           time.Time
+	duration            time.Duration
+	stage               string
+	instrumentationType string
+	category            []string // TODO: switch to using enum?
+	status              Status
+	testSummary         json_schemas.TestSummary
+	targetId            string // TODO: switch to using purl lib?
```

Comment:
Suggestion: let's remove this comment, string is an appropriate representation and parsing it can be done by using a purl lib